src/components/lyne-toast/lyne-toast.tsx

```diff
@@ -0,0 +1,330 @@
+   */
+  private _hasIconSlot: boolean;
+
+  /**
+   * Setup the configuration and prepare the overlay.
+   */
+  public connectedCallback(): void {
+    const defaultConfig: InterfaceToastConfiguration = {
+      ariaLivePoliteness: 'polite',
+      horizontalPosition: 'center',
+      message: null,
+      timeout: 6000,
+      verticalPosition: 'bottom'
+    };
+
+    this._internalConfig = {
+      ...defaultConfig,
+      ...this.config
+    };
+    prepareOverlay(this.el);
+  }
+
+  /**
+   * Evaluate if the icon slot is provided by consumers.
+   */
+  public componentWillLoad(): void {
+    this._hasIconSlot = Boolean(this.el.querySelector('[slot="icon"]'));
+  }
+
+  /**
+   * Present the toast overlay after it has been created.
+   */
+  @Method()
+  public async present(): Promise<void> {
+    await present(this, toastEnterAnimation);
+    if (this._internalConfig.timeout > 0) {
+      this._durationTimeout = setTimeout(() => this.dismiss(undefined, 'timeout'), this._internalConfig.timeout);
+    }
+  }
+
+  /**
+   * Dismiss the toast overlay after it has been presented.
+   *
+   * @param data Any data to emit in the dismiss events.
+   * @param role The role of the element that is dismissing the toast.
+   * Example:
+   * `"cancel"` for close icon
+   * `"timeout"` for auto-close
+   */
+  @Method()
+  public dismiss(data?: any, role?: string): Promise<boolean> {
+    if (this._durationTimeout) {
+      clearTimeout(this._durationTimeout);
+    }
+
+    return dismiss(this, data, role, toastLeaveAnimation);
+  }
+
+  /**
+   * Returns a promise that resolves when the toast did dismiss.
+   */
+  @Method()
+  public onDidDismiss<T = any>(): Promise<InterfaceOverlayEventDetail<T>> {
+    return eventMethod(this.el, 'lyne-toast_did-dismiss');
+  }
+
+  /**
+   * Returns a promise that resolves when the toast will dismiss.
+   */
+  @Method()
+  public onWillDismiss<T = any>(): Promise<InterfaceOverlayEventDetail<T>> {
+    return eventMethod(this.el, 'lyne-toast_will-dismiss');
+  }
+
+  /**
+   * Renders the action/icon button.
+   * @param onClickFn Callback on button click.
+   * @param innerToastEl The HTML rendered as button content.
+   * @returns The button that will be rendered as action/icon.
+   */
+  private _renderActionCommonButton(onClickFn: () => Promise<boolean>, innerToastEl: JSX.Element): JSX.Element {
+    return (
+      <button type='button' tabIndex={0} onClick={onClickFn} role={this._internalConfig.action.role} class={this._buttonClass()}>
+        {innerToastEl}
+      </button>
+    );
+  }
+
+  /**
+   * Creates the CSS classes for the action button.
+   * @returns A CssClassMap with two default CSS classes and all the CSS classes
+   * added by consumers in _internalConfig.action.cssClass.
+   */
+  private _buttonClass(): CssClassMap {
+    const fn: (acc: CssClassMap, curr: string) => CssClassMap = (acc: CssClassMap, curr: string) => {
+      acc[curr] = true;
+
+      return acc;
+    };
```

Comment:
Can't we simplify that to Object.fromEntries ?

Reply:
I think I didn't get it.. 
Basically, I want to convert the classArray into an object like
` {element1OfTheArray: true, element2OfTheArray: true, ...}`
Maybe I can do something like this:

```
const obj = {}
classArray.forEach(e => obj[e] = true);
return { 'lyne-focusable': true, 'toast-button': true, ...obj}
```

but I think the reduce fn does pretty the same..